client/asset/btc/electrum/network.go

```diff
@@ -0,0 +1,646 @@
+	HashFunc string                       `json:"hash_function"`     // e.g. sha256
+	// Services []string                     `json:"services,omitempty"` // e.g. ["tcp://host.com:51001", "ssl://host.com:51002"]
+}
+
+// Features requests the features claimed by the server. The caller should check
+// the Genesis hash field to ensure it is the intended network.
+func (sc *ServerConn) Features(ctx context.Context) (*ServerFeatures, error) {
+	var feats ServerFeatures
+	err := sc.Request(ctx, "server.features", nil, &feats)
+	if err != nil {
+		return nil, err
+	}
+	return &feats, nil
+}
+
+// PeersResult represents the results of a peers server request.
+type PeersResult struct {
+	Addr  string // IP address or .onion name
+	Host  string
+	Feats []string
+}
+
+// Peers requests the known peers from a server (other servers). See
+// SSLPeerAddrs to assist parsing useable peers.
+func (sc *ServerConn) Peers(ctx context.Context) ([]*PeersResult, error) {
+	// [["ip", "host", ["featA", "featB", ...]], ...]
+	// [][]interface{}{string, string, []interface{}{string, ...}}
+	var resp [][]interface{}
+	err := sc.Request(ctx, "server.peers.subscribe", nil, &resp) // not really a subscription!
+	if err != nil {
+		return nil, err
+	}
+	peers := make([]*PeersResult, 0, len(resp))
+	for _, peer := range resp {
+		if len(peer) != 3 {
+			fmt.Printf("bad peer data: %v (%T)", peer, peer)
+			continue
+		}
+		addr, ok := peer[0].(string)
+		if !ok {
+			fmt.Printf("bad peer IP data: %v (%T)", peer[0], peer[0])
+			continue
+		}
+		host, ok := peer[1].(string)
+		if !ok {
+			fmt.Printf("bad peer hostname: %v (%T)", peer[1], peer[1])
+			continue
+		}
+		featsI, ok := peer[2].([]interface{})
+		if !ok {
+			fmt.Printf("bad peer feature data: %v (%T)", peer[2], peer[2])
+			continue
+		}
+		feats := make([]string, len(featsI))
+		for i, featI := range featsI {
+			feat, ok := featI.(string)
+			if !ok {
+				fmt.Printf("bad peer feature data: %v (%T)", featI, featI)
+				continue
+			}
+			feats[i] = feat
+		}
+		peers = append(peers, &PeersResult{
+			Addr:  addr,
+			Host:  host,
+			Feats: feats,
+		})
+	}
+	return peers, nil
+}
+
+// SSLPeerAddrs filters the peers slice and returns the addresses in a
+// "host:port" format in separeate slices for SSL-enabled servers and optionally
+// TCP-only hidden services (.onion host names). Note that if requesting to
+// include onion hosts, the SSL slice may include onion hosts that also use SSL.
+func SSLPeerAddrs(peers []*PeersResult, includeOnion bool) (ssl, tcpOnlyOnion []string) {
+peerloop:
+	for _, peer := range peers {
+		isOnion := strings.HasSuffix(peer.Addr, ".onion")
+		if isOnion && !includeOnion {
+			continue
+		}
+		var tcpOnion string // host to accept if no ssl
+		for _, feat := range peer.Feats {
+			if len(feat) < 2 {
+				continue
+			}
+			switch []rune(feat)[0] {
+			case 't':
+				if !isOnion {
+					continue
+				}
+				port := feat[1:]
```

Comment:
Good point.  I think I might make a comment that we are going to ignore those.  In my observations it is always set, so it shouldn't limit the server selection much.  The reason I'm hesitant to define a default port is because the default is both ill-defined (seeming to be organic convention but not documented) and different between assets.